### PR TITLE
fix(shwap/bitswap): close acccessor in Blockstore

### PR DESCRIPTION
**Pull request by Wondertan**
Blockstore must cleanup the accesor

### CODE REVIEW

File: share/shwap/p2p/bitswap/block_fetch_test.go
Comment (Wondertan) [inline share/shwap/p2p/bitswap/block_fetch_test.go:120]: That is exactly why I don't like the `eds.WithCloser` approach @walldiss

**Comment by walldiss** [comment]
Unfortunately would need rebase after merge of streaming.

File: share/shwap/p2p/bitswap/block_store.go
Comment (walldiss) [inline share/shwap/p2p/bitswap/block_store.go:45]: We now have handy utils.CloseAndLog for that, as it becomes common code accross the repo

Comment (Wondertan) [reply share/shwap/p2p/bitswap/block_store.go:45]: Its not merged to shwap yet and i dont find it useful honestly. I am gonna keep it as it is here

Comment (Wondertan) [reply share/shwap/p2p/bitswap/block_store.go:45]: The log message should contain the context of where the warn/error happens. Using this util will remove any context

Comment (walldiss) [reply share/shwap/p2p/bitswap/block_store.go:45]: It takes logger as first argument, so you can pass as much additional context information as you want within loggier fields.

Comment (walldiss) [reply share/shwap/p2p/bitswap/block_store.go:45]: This is optional, so  would defer to you to decide.

**Review by walldiss — APPROVED** [review]

File: share/shwap/p2p/bitswap/block_store.go
Comment (Wondertan) [reply share/shwap/p2p/bitswap/block_store.go:45]: i dont see any value so gonna keep the way it is

**Pull request merged**
